### ISSUE TITLE
[Content] : Empty Tag field fails to save

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#19189]
Reported via Support: [102228](https://dotcms.zendesk.com/agent/tickets/102228)
Reproducible as of dotCMS 5.3.4.1 and above

The system is failing to create a Contentlet when its Content Type contains a non-required **Tag** field, and no value is set for it.

**To Reproduce**
1. Go to **Content Model > Content Types** and create a new Content Type called "My Test" with the following fields:
1.1 Text field called "Name" / User Searchable / Show in List
1.2 Tag field called "Categories"
1.3 Binary field called "Picture" / System Indexed
1.3 Binary field called "Picture Webp" / System Indexed
2. Go to **Content > Search** and create a Contentlet of type "My Test"
3. Enter a value for the "Name" field **ONLY**. Click **Save**
4. The error being logged is extensive. The main stack traces are:
```
Caused by: java.lang.NullPointerException
	at com.dotmarketing.portlets.categories.business.CategoryFactoryImpl.setParents(CategoryFactoryImpl.java:436) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.portlets.categories.business.CategoryAPIImpl.setParents_aroundBody60(CategoryAPIImpl.java:565) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.portlets.categories.business.CategoryAPIImpl$AjcClosure61.run(CategoryAPIImpl.java:1) ~[dotcms_master_ed3fdfa.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.db.LocalTransaction.wrapReturnWithListeners(LocalTransaction.java:59) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.aspects.interceptors.WrapInTransactionMethodInterceptor.invoke(WrapInTransactionMethodInterceptor.java:23) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.aspects.aspectj.WrapInTransactionAspect.invoke(WrapInTransactionAspect.java:41) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.portlets.categories.business.CategoryAPIImpl.setParents(CategoryAPIImpl.java:528) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.moveContentCategories(ESContentletAPIImpl.java:5625) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.moveContentDependencies(ESContentletAPIImpl.java:5452) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.internalCheckin(ESContentletAPIImpl.java:4754) ~[dotcms_master_ed3fdfa.jar:?]
	... 130 more
11:09:16.967  ERROR actionlet.SaveContentAsDraftActionlet - null
com.dotmarketing.exception.DotRuntimeException: null
	at com.dotcms.exception.ExceptionUtil.bubbleUpException(ExceptionUtil.java:165) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.checkin_aroundBody144(ESContentletAPIImpl.java:4177) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl$AjcClosure145.run(ESContentletAPIImpl.java:1) ~[dotcms_master_ed3fdfa.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.db.LocalTransaction.wrapReturnWithListeners(LocalTransaction.java:59) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.aspects.interceptors.WrapInTransactionMethodInterceptor.invoke(WrapInTransactionMethodInterceptor.java:23) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.aspects.aspectj.WrapInTransactionAspect.invoke(WrapInTransactionAspect.java:41) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.checkin(ESContentletAPIImpl.java:4128) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.checkin(ESContentletAPIImpl.java:4087) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.saveDraft_aroundBody200(ESContentletAPIImpl.java:7949) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl$AjcClosure201.run(ESContentletAPIImpl.java:1) ~[dotcms_master_ed3fdfa.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.db.LocalTransaction.wrapReturnWithListeners(LocalTransaction.java:59) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.aspects.interceptors.WrapInTransactionMethodInterceptor.invoke(WrapInTransactionMethodInterceptor.java:23) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.aspects.aspectj.WrapInTransactionAspect.invoke(WrapInTransactionAspect.java:41) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.saveDraft(ESContentletAPIImpl.java:7941) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.portlets.contentlet.business.ContentletAPIInterceptor.saveDraft(ContentletAPIInterceptor.java:2407) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.portlets.workflows.actionlet.SaveContentAsDraftActionlet.executeAction(SaveContentAsDraftActionlet.java:121) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl.fireWorkflowPostCheckin_aroundBody166(WorkflowAPIImpl.java:2317) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl$AjcClosure167.run(WorkflowAPIImpl.java:1) ~[dotcms_master_ed3fdfa.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.db.LocalTransaction.wrapReturnWithListeners(LocalTransaction.java:59) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.aspects.interceptors.WrapInTransactionMethodInterceptor.invoke(WrapInTransactionMethodInterceptor.java:23) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.aspects.aspectj.WrapInTransactionAspect.invoke(WrapInTransactionAspect.java:41) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl.fireWorkflowPostCheckin(WorkflowAPIImpl.java:2297) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl.lambda$64(WorkflowAPIImpl.java:3014) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.util.ThreadContextUtil.wrapVoidNoReindex(ThreadContextUtil.java:85) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl.fireContentWorkflow_aroundBody180(WorkflowAPIImpl.java:3014) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl$AjcClosure181.run(WorkflowAPIImpl.java:1) ~[dotcms_master_ed3fdfa.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.db.LocalTransaction.wrapReturnWithListeners(LocalTransaction.java:59) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.aspects.interceptors.WrapInTransactionMethodInterceptor.invoke(WrapInTransactionMethodInterceptor.java:23) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotcms.aspects.aspectj.WrapInTransactionAspect.invoke(WrapInTransactionAspect.java:41) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl.fireContentWorkflow(WorkflowAPIImpl.java:2991) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.portlets.contentlet.business.web.ContentletWebAPIImpl.saveWebAsset(ContentletWebAPIImpl.java:487) ~[dotcms_master_ed3fdfa.jar:?]
	at com.dotmarketing.portlets.contentlet.business.web.ContentletWebAPIImpl.saveContent_aroundBody0(ContentletWebAPIImpl.java:135) ~[dotcms_master_ed3fdfa.jar:?]
```

**Expected behavior**
A non-required field should never cause the content creation to fail if no value is passed down.

**Comment by jcastro-dotcms** [comment]
PR: https://github.com/dotCMS/core/pull/19191

**Comment by erickgonzalez** [comment]
Working as it should, able to save content with/without tags.

**Comment by bryanboza** [comment]
Fixed tested on release-5.3.8 // Postgres // FF
